odl/trafos/wavelet.py

```diff
@@ -106,37 +111,163 @@ def coeff_size_list(shape, nscales, wbasis, mode):
         ``N`` = number of scaling levels = nscales
     """
     if len(shape) not in (1, 2, 3):
-        raise ValueError('shape must have length 1, 2 or 3, got {}.'
+        raise ValueError('Shape must have length 1, 2 or 3, got {}.'
                          ''.format(len(shape)))
 
-    max_level = pywt.dwt_max_level(shape[0], filter_len=wbasis.dec_len)
-    if nscales > max_level:
-        raise ValueError('Too many scaling levels, got {}, maximum useful'
-                         ' level is {}'
-                         ''.format(nscales, max_level))
-
-    # dwt_coeff_len calculates the number of coefficients at the next
-    # scaling level given the input size, the length of the filter and
-    # the applied mode.
-    # We use this in the following way (per dimension):
-    # - length[0] = original data length
-    # - length[n+1] = dwt_coeff_len(length[n], ...)
-    # - until n = nscales
-    size_list = [shape]
-    for scale in range(nscales):
-        shp = tuple(pywt.dwt_coeff_len(n, filter_len=wbasis.dec_len, mode=mode)
-                    for n in size_list[scale])
-        size_list.append(shp)
-
-    # Add a duplicate of the last entry for the approximation coefficients
-    size_list.append(size_list[-1])
-
-    # We created the list in reversed order compared to what pywt expects
-    size_list.reverse()
+    if nscales is None and axes is None:
+        raise ValueError('Either nscales or axes has to be defined')
+
+    if axes is None:
```

Comment:
It's more readable if you structure is like this
```python
if axes is not None:
    if nscales is not None:
        raise ValueError(...)
    # Do something with axes
elif nscales is not None:
    # Do something with nscales
else:
    raise ValueError('either `nscales` or `axes` has to be given')
```
You are using opposite logic, i.e. check if something is *not* given, and then do the other thing, which makes the code more non-obvious.